src/core/style/_elevation.scss

```diff
@@ -0,0 +1,186 @@
+  19: '0px 7px 36px 6px #{$_ambient-color}',
+  20: '0px 8px 38px 7px #{$_ambient-color}',
+  21: '0px 8px 40px 7px #{$_ambient-color}',
+  22: '0px 8px 42px 7px #{$_ambient-color}',
+  23: '0px 9px 44px 8px #{$_ambient-color}',
+  24: '0px 9px 46px 8px #{$_ambient-color}'
+);
+
+/**
+ * The css property used for elevation. In most cases this should not be changed. It is exposed
+ * as a variable for abstraction / easy use when needing to reference the property directly, for
+ * example in a `will-change` rule.
+ */
+$md-elevation-property: box-shadow !default;
+
+/**
+ * The default duration value for elevation transitions.
+ */
+$md-elevation-transition-duration: 280ms !default;
+
+/**
+ * The default easing value for elevation transitions.
+ */
+$md-elevation-transition-timing-function: $md-fast-out-slow-in-timing-function;
+
+/**
+ * Applies the correct css rules to an element to give it the elevation specified by $zValue.
+ * The $zValue must be between 0 and 24.
+ */
+@mixin md-elevation($zValue) {
+  @if type-of($zValue) != number or not unitless($zValue) {
+    @error "$zValue must be a unitless number";
+  }
+  @if $zValue < 0 or $zValue > 24 {
+    @error "$zValue must be between 0 and 24";
+  }
+
+  #{$md-elevation-property}: #{map-get($_umbra-elevation-map, $zValue)},
+                             #{map-get($_penumbra-elevation-map, $zValue)},
+                             #{map-get($_ambient-elevation-map, $zValue)};
+}
+
+/**
+ * Returns a string that can be used as the value for a `transition` property for elevation.
+ * Calling this function directly is useful in situations where a component needs to transition
+ * more than one property.
+ *
+ * ```scss
+ * .foo {
+ *   transition: md-elevation-transition-rule(), opacity 100ms ease;
+ *   will-change: $md-elevation-property, opacity;
+ * }
+ * ```
+ */
+@function md-elevation-transition-rule(
+    $duration: $md-elevation-transition-duration,
+    $easing: $md-elevation-transition-timing-function) {
+  @return #{$md-elevation-property} #{$duration} #{$easing};
+}
+
+/**
+ * Applies the correct css rules needed to have an element transition between elevations.
+ * This mixin should be applied to elements whose elevation values will change depending on their
+ * context (e.g. when active or disabled).
+ */
+// NOTE(traviskaufman): Both this mixin and the above function use default parameters so they can
+// be used in the same way by clients.
+@mixin md-elevation-transition(
+    $duration: $md-elevation-transition-duration,
+    $easing: $md-elevation-transition-timing-function) {
+  transition: md-elevation-transition-rule($duration, $easing);
+  will-change: $md-elevation-property;
+}
+
+// Provides external CSS classes for each elevation value. Each CSS class is formatted as
+// `md-elevation-z$zValue` where `$zValue` correspond to the z-space in which the element is
+// elevated to.
+@for $zValue from 0 through 24 {
+  .md-elevation-z#{$zValue} {
```

Comment:
In Material 1, we deprecated the `z` suffix, and decided to use `dp`. Is here another background?